platform/android/java_godot_wrapper.cpp

```diff
@@ -105,12 +98,16 @@ GodotJavaWrapper::~GodotJavaWrapper() {
 	ERR_FAIL_NULL(env);
 	env->DeleteGlobalRef(godot_instance);
 	env->DeleteGlobalRef(godot_class);
-	env->DeleteGlobalRef(activity);
-	env->DeleteGlobalRef(activity_class);
 }
 
 jobject GodotJavaWrapper::get_activity() {
-	return activity;
+	if (_get_activity) {
+		JNIEnv *env = get_jni_env();
+		ERR_FAIL_NULL_V(env, nullptr);
+		jobject activity = env->CallObjectMethod(godot_instance, _get_activity);
+		return activity;
+	}
+	return nullptr;
```

Comment:
I think that getting this on demand means that anything that calls `GodotJavaWrapper::get_activity()` will need to remember to `env->DeleteLocalRef(activity)` when it's done?

This may also create some issues with `DisplayServerAndroid::window_get_native_handle(WINDOW_HANDLE)` which will return a `jobject` that may not be valid after it's been returned